config/routes.rb

```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
-  get 'pages/post'
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show'
+  get '/users/:user_id/posts', to: 'posts#index'
+  get '/users/:user_id/posts/:id', to: 'posts#show'
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
 end
```

Comment:
- [ ] I suggest that you use `resources` because Resource routing allows you to quickly declare all of the common routes for a given resourceful controller and it makes your codes clean. you can do something like
`resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end`